app/experimenter/nimbus-ui/src/components/AppLayoutSidebarLocked/index.tsx

```diff
@@ -44,20 +42,11 @@ const getSidebarItems = (
   return sidebarItems;
 };
 
-const probesetToMapping = (
-  probesets:
-    | (getExperiment_experimentBySlug_primaryProbeSets | null)[]
-    | (getExperiment_experimentBySlug_secondaryProbeSets | null)[],
-) => {
-  const newMap: { [key: string]: string } = {};
-  probesets.map(
-    (
-      probeset:
-        | (getExperiment_experimentBySlug_primaryProbeSets | null)
-        | (getExperiment_experimentBySlug_secondaryProbeSets | null),
-    ) => (newMap[probeset!.slug] = probeset!.name),
-  );
-  return newMap;
+const outcomeToMapping = (outcomes: OutcomesList) => {
+  return outcomes.reduce((acc: { [key: string]: string }, outcome) => {
+    acc[outcome?.slug as string] = outcome?.friendlyName!;
+    return acc;
+  }, {});
```

Comment:
When I tried adding the type import locally to `react-app-env.d.ts`, I saw some type errors hereabouts. Did some code golfing and came up with this:
```javascript
import { getConfig_nimbusConfig_outcomes } from "../../types/getConfig";

const outcomeToMapping = (outcomes: OutcomesList) =>
  outcomes
    .filter(
      (outcome): outcome is getConfig_nimbusConfig_outcomes =>
        !!outcome && !!outcome.slug,
    )
    .reduce(
      (acc, { slug, friendlyName }) => ({
        ...acc,
        [slug!]: friendlyName,
      }),
      {},
    );
```
Not sure if all those gymnastics to guard against nulls & undefineds will prevent future issues, but typescript seemed to like this better?

Reply:
Oh yeah this broke all the types being used in the declaration file. I'll move the outcome types to an importable file.

![Screen Shot 2021-03-10 at 10 41 31 AM](https://user-images.githubusercontent.com/6392049/110655609-3553fa80-818d-11eb-9413-36ca5feaa0a1.png)